src/lib/interactionManager/authorizationFlow.ts

```diff
@@ -0,0 +1,56 @@
+import { Flow } from './flow'
+import {
+  FlowType,
+  AuthorizationResponse,
+  AuthorizationRequest,
+  AuthorizationType,
+  AuthorizationFlowState,
+} from './types'
+import { isAuthorizationRequest, isAuthorizationResponse } from './guards'
+
+export class AuthorizationFlow extends Flow<
+  AuthorizationResponse | AuthorizationRequest
+> {
+  public type = FlowType.Authorization
+  public state: AuthorizationFlowState = {
+    description: '',
+    imageURL: '',
+    action: '',
+  }
+
+  public handleInteractionToken(
+    token: AuthorizationRequest | AuthorizationResponse,
+    interactionType: string,
+  ) {
+    switch (interactionType) {
+      case AuthorizationType.AuthorizationRequest:
+        if (isAuthorizationRequest(token))
+          return this.consumeAuthorizationRequest(token as AuthorizationRequest)
```

Comment:
Is the `as` statement required given the type guard above?

Reply:
Fixed.